core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -0,0 +1,1347 @@
+            .blob_verification_proof
+            .batch_medatada
+            .batch_header
+            .batch_root;
+        let blob_index = cert.blob_verification_proof.blob_index;
+        let blob_header = &cert.blob_header;
+
+        let blob_header_hash = self.hash_encode_blob_header(blob_header);
+        let leaf_hash = web3::keccak256(&blob_header_hash);
+
+        let generated_root =
+            self.process_inclusion_proof(inclusion_proof, leaf_hash, blob_index)?;
+
+        if generated_root != *root {
+            return Err(VerificationError::DifferentRoots {
+                expected: hex::encode(root),
+                actual: hex::encode(&generated_root),
+            });
+        }
+        Ok(())
+    }
+
+    fn hash_batch_metadata(
+        &self,
+        batch_header: &BatchHeader,
+        signatory_record_hash: &[u8],
+        confirmation_block_number: u32,
+    ) -> Vec<u8> {
+        let batch_header_token = Token::Tuple(vec![
+            Token::FixedBytes(batch_header.batch_root.clone()), // Clone only where necessary
+            Token::Bytes(batch_header.quorum_numbers.clone()),
+            Token::Bytes(batch_header.quorum_signed_percentages.clone()),
+            Token::Uint(ethabi::Uint::from(batch_header.reference_block_number)),
+        ]);
+
+        let encoded = encode(&[batch_header_token]);
+        let header_hash = web3::keccak256(&encoded).to_vec();
+
+        let hash_token = Token::Tuple(vec![
+            Token::FixedBytes(header_hash.to_vec()),
+            Token::FixedBytes(signatory_record_hash.to_owned()), // Clone only if required
+        ]);
+
+        let mut hash_encoded = encode(&[hash_token]);
+
+        hash_encoded.append(&mut confirmation_block_number.to_be_bytes().to_vec());
+        web3::keccak256(&hash_encoded).to_vec()
+    }
+
+    /// Retrieves the block to make the request to the service manager
+    async fn get_context_block(&self) -> Result<u64, VerificationError> {
+        let latest = self
+            .signing_client
+            .as_ref()
+            .block_number()
+            .await
+            .map_err(ServiceManagerError::EnrichedClient)?
+            .as_u64();
+
+        let depth = self
+            .cfg
+            .settlement_layer_confirmation_depth
+            .saturating_sub(1);
+        let block_to_return = latest.saturating_sub(depth as u64);
+        Ok(block_to_return)
+    }
+
+    async fn call_batch_id_to_metadata_hash(
+        &self,
+        blob_info: &BlobInfo,
+    ) -> Result<Vec<u8>, VerificationError> {
+        let context_block = self.get_context_block().await?;
+
+        let func_selector =
+            ethabi::short_signature("batchIdToBatchMetadataHash", &[ParamType::Uint(32)]);
+        let mut data = func_selector.to_vec();
+        let mut batch_id_vec = [0u8; 32];
+        U256::from(blob_info.blob_verification_proof.batch_id).to_big_endian(&mut batch_id_vec);
+        data.append(batch_id_vec.to_vec().as_mut());
```

Comment:
Please use `ethabi::encode()` instead of this; this would make the code more transparent.

Reply:
Fixed here: https://github.com/lambdaclass/zksync-era/pull/386